README.md

```diff
@@ -166,6 +166,7 @@ DbConfig:
 DeployConfig:
   EnvironmentType: dev 
 EnsemblingJobConfig:
+  Enabled: true
```

Comment:
I brought this point during the orchestration PR review, but here it is relevant again: can we come up with a better structure of the batch ensembling related configuration? Here `Enabled` is under `EnsemblingJobConfig`, but there is also `BatchRunnerConfig` which feels to be even better for this feature toggle. Would it be better to have a single top-level config branch (for instance `BatchEnsembling`) and nest all related configuration under it:

```yaml
BatchEnsembling:
  Enabled: true

  RunnerConfig:
    TimeInterval: 10s
    # ...

  JobConfig:
    DefaultEnvironment: dev
    # ...
```

Reply:
Also `RecordsToProcessInOneIteration` and (maybe) `MaxRetryCount` seem to be better defined under `BatchRunnerConfig` rather than under `EnsemblingJobConfig` because they define the behaviour of the runner, not the job iteslf.

Reply:
Yeah Ill take some time to rearrange and review the configuration

Reply:
A consequence of the proposed configuration is that every BatchRunner has it's own time interval between runs, I'm not sure if that level of granuality is required.

I would propose either one of the following (group by runners or group by batch job). Option 1 would be similar to your suggestion but I would prefer option 2 because of the grouping by runners. It is probably neater to group by runner. But I do not have a strong opinion on this. @romanwozniak do let me know which option you prefer/other suggestions.

1. Group by batch job
```yaml
BatchEnsembling:
  Enabled: true
  
  # Job Config -> common properties
  JobConfig:
    DefaultEnvironment: dev
  
  # Runner configuration -> corresponds to runner.go
  RunnerConfig:
    TimeInterval: 10s
    RecordsToProcessInOneIteration: 10
    MaxRetryCount: 3
    
  # ImageBuilder configuration -> corresponds to the imagebuilder package
  ImageBuilderConfig:
    ImageConfig:
      Registry: ghcr.io
      BaseImageRef: ghcr.io/gojek/turing/batch-ensembler:latest
      BuildNamespace: default
      BuildContextURI: git://github.com/gojek/turing.git#refs/heads/main
      DockerfileFilePath: engines/batch-ensembler/app.Dockerfile
      BuildTimeoutDuration: 20m
    KanikoConfig:
      Image: gcr.io/kaniko-project/executor
      ImageVersion: v1.5.2
      ResourceRequestsLimits:
        Requests:
          CPU: "1"
          Memory: 1Gi
        Limits:
          CPU: "1"
          Memory: 1Gi
```

2. Group by runner.
```yaml
BatchRunnerConfig:
  TimeInterval: 10s

  BatchEnsembling:
    Enabled: true
    
    # Job Config -> common properties
    JobConfig:
      DefaultEnvironment: dev

    # RunnerConfig configuration -> corresponds to the runner.go
    RunnerConfig:
      RecordsToProcessInOneIteration: 10
      MaxRetryCount: 3
      
    # ImageBuilder configuration -> corresponds to imagebuilder.go
    ImageBuilderConfig:
      ImageConfig:
        Registry: ghcr.io
        BaseImageRef: ghcr.io/gojek/turing/batch-ensembler:latest
        BuildNamespace: default
        BuildContextURI: git://github.com/gojek/turing.git#refs/heads/main
        DockerfileFilePath: engines/batch-ensembler/app.Dockerfile
        BuildTimeoutDuration: 20m
      KanikoConfig:
        Image: gcr.io/kaniko-project/executor
        ImageVersion: v1.5.2
        ResourceRequestsLimits:
          Requests:
            CPU: "1"
            Memory: 1Gi
          Limits:
            CPU: "1"
            Memory: 1Gi

  Foo:
    Enabled: false

  Bar:
    Enabled: false
```

Reply:
Option 2 is interesting, but seems like a premature optimization to me. Why would we need to have multiple runners with different configuration? Also, I'd prefer to have a feature toggle as a root level flag, so it can on/off the entire batch ensembling component. With option 2 you rather enable runners, and not the feature itself

Reply:
Actually option 2 has less abstraction as it groups up on a runner level; it assumes no similarity between runners other than they are able to turn off and on individually.

I think there is some confusion on terminology here, let me set the foundations before we miscommunicate.
Batch Runner -> Runs mutiple batch runners. (i.e. batch ensembling job is one of the many runners that can be configured.)
Batch Ensembling job -> Is a batch runner. The job that runs the batch ensember.

So in option 2 you can turn off a feature, i.e. BatchEnsembling or Foo or Bar. This is useful because:
1. If we were to add more, we can toggle individually. i.e. turn off Batch Ensembling and not Foo batch feature, not a turn off all or turn on all runners.
2. Common interface for all runners -> users don't have to relearn a new pattern for the configuration.

Reply:
The main reason I'm not happy with option 2 is because I can't think of any "other" Foo or Bar runners in a foreseeable future, hence that's the premature optimization that I'd like to avoid

Reply:
Alright I think that is fair. Option 1 it is.